api/v1alpha1/virtualmachineimage_types.go

```diff
@@ -95,6 +101,9 @@ type VirtualMachineImageStatus struct {
 	// Deprecated
 	PowerState string `json:"powerState,omitempty"`
 
+	// ImageName describes the display name of this VirtualMachineImage.
+	ImageName string `json:"imageName"`
```

Comment:
I think this should be //+optional and omitempty.

Reply:
We use the ImageName to generate the VirtualMachineImage object name in the cluster, why would this field be optional?

Reply:
We should do so that these fields are Patch'able.